jcl/src/java.management/share/classes/java/lang/management/ManagementFactory.java

```diff
@@ -466,51 +466,55 @@ public static ThreadMXBean getThreadMXBean() {
 		return ManagementUtils.getPlatformManagementInterfaces();
 	}
 
-	private static final class ServerHolder {
+	private static final class ServerHolder implements PrivilegedAction<MBeanServer> {
 
-		static final MBeanServer platformServer;
+		private static final ServerHolder instance = new ServerHolder();
 
-		static {
-			platformServer = MBeanServerFactory.createMBeanServer();
+		private MBeanServer platformServer;
```

Comment:
Shouldn't this be `volatile` to ensure coherence?

Reply:
Alternatively, using an enum to create the singleton is the approach advocated in Effective Java

Reply:
Initialization is synchronized so volatile is not needed.
I believe using an enum would yield the same behavior this is intended to fix.

Reply:
Unless its volatile, there is no guaranteed happens-before relationship between allocation and assignment (unless there are final fields in the object).

Reply:
I will make that field `volatile`.